WordPress/gutenberg
# [Mobile] Update media placeholders and block outline

**Related PRs:**
- https://github.com/wordpress-mobile/gutenberg-mobile/pull/5969
- https://github.com/wordpress-mobile/WordPress-iOS/pull/21119
- https://github.com/wordpress-mobile/WordPress-Android/pull/18788

## What?
This PR brings back the BlockOutline that was removed in https://github.com/WordPress/gutenberg/pull/51222 and also updates the MediaPlaceholder with new styles.

## Why?
To prevent confusion of selected states of blocks and also to improve our current editor UI.

## How?
It restores the `BlockOutline` component and updates its logic so the outline will be shown when the component is selected and meets one of the following conditions: it is a `text` block with inner blocks, it is a block other than `text` with inner blocks and is at root list level, or it is a block other than `text` with inner blocks and is not at root list level.

New colors were added `$blue-60`, `$blue-40` and `$dark-dim`.

## MediaPlaceholder

Updates the component so it can be selected before the whole container had a touch action so it wouldn't be possible to be in a selected state.

Now a button will be shown in the middle to allow users to upload media, it also updates the styles and cleans up the code a bit.

## Embed Placeholder

It brings the same style modifications from the MediaPlaceholder component to this one.

## Tests

General snapshot updates and integration tests were updated after these changes.

## Testing Instructions

## Test case 1
- Open the editor
- Add several media blocks and check the placeholders can be selected and have the new styles (Image, Gallery, Cover, Embed, VideoPress)

## Test case 2
- Add a List block
- Add several list items
- Tap the "Up" arrow to select the parent block
- **Expect** to see the blue outline

## Test case 3

- Add different kinds of blocks
- Tap on them and **expect** to see the block outlines for parent blocks that have inner blocks, or blocks that are not text-based.

### Testing Instructions for Keyboard
N/A

## Screenshots or screencast 
iOS|Android
-|-
<video src="https://github.com/WordPress/gutenberg/assets/4885740/c43f5d5e-78a1-46ff-a26c-f81f54dae094" width=200 />|<video src="https://github.com/WordPress/gutenberg/assets/4885740/b71d3dd4-ef57-4641-b4fc-bd124c107099" width=200 />
